framework/Optimizers/Optimizer.py

```diff
@@ -571,12 +596,40 @@ def updateMultilevelDepth(self, traj, depth, optPoint, setAll=False):
     # clear existing gradient determination data
     if not firstTime:
       self.clearCurrentOptimizationEffort(traj)
+    # apply preconditioner IFF we're going towards INNER loops
+    if depth > oldDepth:
+      self.raiseADebug('Preconditioning subsets below',oldDepth,range(oldDepth+1,depth+1))
+      #apply changes all the way down
+      for d in range(oldDepth+1,depth+1):
+        precondBatch = self.mlSequence[d]
+        precond = self.mlPreconditioners.get(precondBatch,None)
+        if precond is not None:
+          self.raiseADebug('Running preconditioner on batch "{}"'.format(precondBatch))
+          infoDict = {'SampledVars':self.denormalizeData(optPoint)}
+          _,(results,_) = precond.evaluateSample([infoDict['SampledVars']],'Optimizer',infoDict)
+          # flatten results #TODO breaks for multi-entry arrays
+          for key,val in results.items():
+            results[key] = float(val)
+          self.proposeNewPoint(traj,self.normalizeData(results))
+          self.status[traj]['process'] = 'submitting new opt points'
+          self.status[traj]['reason'] = 'received recommended point'
     # if there's batch info about the new batch, set it
     self._setAlgorithmState(traj,self.mlBatchInfo[newBatch].get(traj,None))
     #make sure trajectory is live
     if traj not in self.optTrajLive:
       self.optTrajLive.append(traj)
 
+  def proposeNewPoint(self,traj,point):
+    """
+      Sets a proposed point for the next in the opt chain.  Recommended to be overwritten in subclasses.
+      @ In, traj, int, trajectory who is getting proposed point
+      @ In, point, dict, new input space point as {var:val}
+      @ Out, None
+    """
+    point = copy.deepcopy(point)
+    self.optVarsHist[traj][self.counter['varsUpdate'][traj]] = point
+    self.recommendedOptPoint[traj] = point
```

Comment:
Correct.  Neither entity was designed to change "point", but if either do, it should change in both.